util/src/testing.rs

```diff
@@ -0,0 +1,95 @@
+use core::fmt;
+use core::mem;
+use core::slice;
+
+/// Internal definition type used for a test.
+pub struct Test {
+    pub module: &'static str,
+    pub name: &'static str,
+    pub run: fn() -> bool,
```

Comment:
nit: WDYT about making `run` private, and adding a

```rust
impl Test {
    pub fn run(&self) -> bool {
        (self.run)()
    }
}
```
so that code consuming this can treat it as a method?

Reply:
We can't do that, because then we wouldn't be able to construct this struct from within the `decl_test!` macro.